src/utils/ServiceLocator.cxxtest

```diff
@@ -82,3 +83,27 @@ TEST_F(ServiceLocatorTests, register_via_interface)
     service = ServiceLocator<OtherInterface>::get_service();
     EXPECT_EQ(nullptr, service);
 }
+
+TEST_F(ServiceLocatorTests, register_using_alias)
+{
+    std::shared_ptr<OtherInterface> first(new OtherImplementation1());
+    std::shared_ptr<OtherInterface> second(new OtherImplementation2());
+
+    ServiceLocator<OtherInterface>::register_service(first);
+    shared_ptr<OtherInterface> service = ServiceLocator<OtherInterface>::get_service();
+    ASSERT_NE(nullptr, service);
+    EXPECT_EQ("One", service->get_name());
+
+    // Register second with an alias
+    ServiceLocator<OtherInterface>::register_service(second, "other");
+
+    // Default still returns first
+    service = ServiceLocator<OtherInterface>::get_service();
+    ASSERT_NE(nullptr, service);
+    EXPECT_EQ("One", service->get_name());
+
+    // Alias returns second
+    service = ServiceLocator<OtherInterface>::get_service("other");
+    ASSERT_NE(nullptr, service);
+    EXPECT_EQ("Two", service->get_name());
+}
```

Comment:
It is probably redundant given the get_name(), but should we have a test that the first service pointer != the second service pointer?